server/routes/volunteers.ts

```diff
@@ -241,11 +263,21 @@ router.route("/api/volunteers/update/:volunteerId").patch(
         url: lookupNewNumber,
       });
 
+      if (newNumberFetch.kind === "error") {
+        logger.error(newNumberFetch.error);
+        res.status(INTERNAL_SERVER_ERROR).json({
+          message: newNumberFetch.error,
+        });
+
+        return;
+      }
+
       if (newNumberFetch.records.length > 0) {
-        res.status(BAD_REQUEST);
-        throw new Error(
-          `The phone number ${phoneNumber} is already in use by another volunteer.`
-        );
+        res.status(BAD_REQUEST).json({
+          message: `The phone number ${phoneNumber} is already in use by another volunteer.`,
+        });
```

Comment:
For all these error scenarios, can we also log them to the server log so we have a trail there? Looks like these are just sending the error back to the frontend.

Reply:
I added more logs to address this